### PR TITLE
Documentation updates for 2.2.

### PR DESCRIPTION
**Pull request by nawroth**
* Upgrades.
* Deprecations.
* Javadocs.
* Adds short abstracts to many Cypher pages.
* Restructured the introduction.

### CODE REVIEW

File: community/cypher/docs/cypher-docs/src/docs/dev/ql/match/index.asciidoc [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
 == Introduction ==
 
-The +MATCH+ clause allows you to specify the patterns Cypher will search for in the database, and
-it is the primary way of getting data into the current set of bindings.
+The `MATCH` clause allows you to specify the patterns Neo4j will search for in the database.
+This is the primary way of getting data into the current set of bindings.
 It is worth reading up more on the specification of the patterns themselves in <<introduction-pattern>>.
 
-It is often coupled to a +WHERE+ clause which adds restrictions, or predicates, to the +MATCH+ patterns, making them more specific.
+`MATCH` is often coupled to a `WHERE` clause which adds restrictions, or predicates, to the `MATCH` patterns, making them more specific.
```
Comment (systay) [inline community/cypher/docs/cypher-docs/src/docs/dev/ql/match/index.asciidoc]: Strictly speaking, `WHERE` is not it's own clause

Comment (mneedham) [inline community/cypher/docs/cypher-docs/src/docs/dev/ql/match/index.asciidoc]: @systay  do you think it confuses the mental model the user has if we talk about `MATCH` and `WHERE` as being separate clauses?

Comment (nawroth) [inline community/cypher/docs/cypher-docs/src/docs/dev/ql/match/index.asciidoc]: I think the common term for this would be a _subclause_. WDYT?

**Comment by mneedham** [comment]
Good stuff. I've added some comments with some small tweaks to the wording of some sentences.

File: community/cypher/docs/cypher-docs/src/docs/dev/ql/match/index.asciidoc (hunk 1 above)
Comment (mneedham) [inline community/cypher/docs/cypher-docs/src/docs/dev/ql/match/index.asciidoc]: I like that. What say @systay ?

**Comment by nawroth** [comment]
Thanks for the feedback @mneedham @systay -- I have pushed a new version which should take care of the issues.

File: community/cypher/docs/cypher-docs/src/docs/dev/ql/match/index.asciidoc (hunk 1 above)
Comment (systay) [inline community/cypher/docs/cypher-docs/src/docs/dev/ql/match/index.asciidoc]: :+1:

**Comment by nawroth** [comment]
I'm currently rebasing and fixing conflicts.

**Comment by mneedham** [comment]
Was getting some weird extra merge commits when I tried to directly merge it in so I did a cherry-pick instead - https://github.com/neo4j/neo4j/commit/88db5715e4fc0ef543af5e5bd92545f72d0ead4a

**Pull request closed without merging**